handlers.go

```diff
@@ -488,6 +488,40 @@ func (s *Server) handleAddCar(c echo.Context, u *User) error {
 
 	// TODO: how to specify filename?
 	filename := header.Roots[0].String()
+	if qpname := c.QueryParam("name"); qpname != "" {
+		filename = qpname
+	}
+
+	var commpcid cid.Cid
+	var commpSize uint64
+	if cpc := c.QueryParam("commp"); cpc != "" {
+		if u.Perm < util.PermLevelAdmin {
+			return fmt.Errorf("must be an admin to specify commp for car file upload")
+		}
+
+		sizestr := c.QueryParam("size")
+		if sizestr == "" {
+			return fmt.Errorf("must also specify 'size' when setting commP for car files")
+		}
+
+		ss, err := strconv.ParseUint(sizestr, 10, 64)
+		if err != nil {
+			return fmt.Errorf("failed to parse size: %w", err)
+		}
+
+		commpSize = ss
+
+		cc, err := cid.Decode(cpc)
+		if err != nil {
+			return err
+		}
+
+		if cc.Prefix().Codec != cid.FilCommitmentUnsealed {
+			return fmt.Errorf("can only specify an unsealed commitment as a commP")
+		}
```

Comment:
This entire thing can be replaced by https://pkg.go.dev/github.com/filecoin-project/go-fil-commcid#CIDToDataCommitmentV1 and just proxy the error (if any) in full. It does codc *and* mh checks.